genesis/engine/solvers/rigid/collider_decomp.py

```diff
@@ -1320,12 +1321,41 @@ def _func_mpr(self, i_ga, i_gb, i_b):
                                     # Note that SDF may detect different collision points depending on geometry ordering.
                                     # Because of this, it is necessary to run it twice and take the contact information
                                     # associated with the point of deepest penetration.
-                                    is_col_a, normal_a, penetration_a, contact_pos_a = self._func_contact_vertex_sdf(
-                                        i_ga, i_gb, i_b
+                                    is_col_a, normal_a, penetration_a, contact_pos_a = (
```

Comment:
This is not the right pattern in this case. You don't need to pre-allocate `_a` and `_b`. Just use your `_tmp` and adapt the logic right after, i.e. line 1359:
```python
if i_sdf == 0:
    if is_col_tmp:
        normal, penetration, contact_pos = normal_tmp, penetration_tmp, contact_pos_tmp
else:
    if is_col_tmp and (not is_col or penetration_tmp >= penetration):
        normal, penetration, contact_pos = normal_tmp, penetration_tmp, contact_pos_tmp
```

Reply:
Actually this is buggy because `is_col` is not updated. Here is the right implementation:
```python
for i_sdf in range(2):
    is_col_tmp, normal_tmp, penetration_tmp, contact_pos_tmp = self._func_contact_vertex_sdf(
        i_ga if i_sdf == 0 else i_gb, i_gb if i_sdf == 0 else i_ga, i_b
    )
    if i_sdf == 0:
        if is_col_tmp:
            normal, penetration, contact_pos = normal_tmp, penetration_tmp, contact_pos_tmp
        else:
            is_col = False
    else:
        if is_col_tmp and (not is_col or penetration_tmp >= penetration):
            normal, penetration, contact_pos = normal_tmp, penetration_tmp, contact_pos_tmp
is_col = True
```